get_ndvi_data.R

```diff
@@ -1,39 +1,152 @@
-# Get NDVI data from MODIS
+#Functions for retrieving and processing NDVI data for BBS route locations. Currently only GIMMS is suported, with years 1981-2013 available.
+#
+#GIMMS you just need to call get_bbs_gimms_ndvi() to retreive a dataframe of (site_id, year, month, ndvi). 
+#-It will take a while to download the 14gb of data and extract values the 1st time around. Results are stored in an sqlite db for quick access.
+#-To redo it just delete the database. 
+#-missing values are filled in with -99
 
-library(MODISTools)
+library(rgdal)
+library(gimms)
 library(dplyr)
+library(tidyr)
+library(sp)
+library(raster)
+#library(doParallel)
 
-route_data <- read.csv('data/BBS_routes.csv')
-route_data$site_id <-  route_data$statenum * 1000 + route_data$route
+sqlite_db_file='./data/bbsforecasting.sqlite'
+database <- src_sqlite(sqlite_db_file, create = TRUE)
+gimms_folder='./data/gimms_ndvi/'
+dir.create(gimms_folder, showWarnings = FALSE, recursive = TRUE)
+years_to_use=1981:2014
 
-contig_modern_routes <- read.csv('data/contig_modern_routes.csv')
+#################################################
+#This assumes we want all gimms files that are available. It queries for files
+#that are available for download and checks to see which, if any, are missing.
+#Returns TRUE if all files that are available are already present, and list of 
+#files to download if any are missing.
+##################################################
+check_if_gimms_files_present=function(){
+  available_files_download_path=sort(gimms::updateInventory())
+  available_files_name=basename(available_files_download_path)
+  
+  files_present=sort(list.files(gimms_folder))
+  #hdr files are created from some of the gimms processing that we don't want to
+  #use here.
+  files_present=files_present[!grepl('hdr', files_present)]
+  
+  if(length(files_present)>0){
+    to_download=available_files_download_path[! available_files_name == files_present]
+  } else {
+    to_download=available_files_download_path
+  }
+  
+  if(length(to_download)==0){
+    return(TRUE)
+  } else {
+    return(to_download)
+  }
```

Comment:
What about just returning `to_download` here regardless and then checking for `NULL` instead of `TRUE` when the function is called?